### PR TITLE
WIP: Discord integration structure

### PR DESCRIPTION
**Pull request by Electroid**
Work in progress, will update periodically then promote to a PR.

### CODE REVIEW

File: src/main/java/tc/oc/pgm/datastore/DatastoreImpl.java [hunk 1]
```diff
@@ -367,47 +363,34 @@ private void updateDiscordUser(UUID id, @Nullable Long snowflake) throws SQLExce
...
-          valid = true;
-        } catch (SignatureException e) {
-          logger.log(Level.WARNING, "Unable to create one-time-pin", e);
-          pin = Double.toString(new Random().nextGaussian());
-          valid = false;
-        }
-      }
+      this.pin =
+          random.ints(2, 100, 1000).mapToObj(Integer::toString).collect(Collectors.joining("-"));
+      this.expires = System.currentTimeMillis() + Duration.ofMinutes(5).toMillis();
```
Comment (Pablete1234) [inline src/main/java/tc/oc/pgm/datastore/DatastoreImpl.java]: Why not use an `Instant` here?
`this.expires = Instant.now().plusMinutes(5);`

File: src/main/java/tc/oc/pgm/discord/DiscordServerImpl.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  }
+
+  @Override
+  public void onPrivateMessageReceived(PrivateMessageReceivedEvent event) {
+    final String message = event.getMessage().getContentStripped();
+    final Matcher matcher = COMMAND.matcher(message);
+
+    if (matcher.matches()
+        && requestVerification(event.getAuthor().getIdLong(), matcher.group(1), matcher.group(2))) {
+      event.getMessage().addReaction("✅").submit();
```
Comment (TheMolkaPL) [inline src/main/java/tc/oc/pgm/discord/DiscordServerImpl.java]: Can we make it a constant?

File: src/main/java/tc/oc/pgm/discord/DiscordServerImpl.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    }
+  }
+
+  @Override
+  public DiscordUser getBot() {
+    if (client == null) return null;
```
Comment (TheMolkaPL) [inline src/main/java/tc/oc/pgm/discord/DiscordServerImpl.java]: `IllegalStateException`?

**Comment by Electroid** [comment]
I'm closing this for now, not being prioritized, but branch will still exist.

**Pull request closed without merging**